linux/mprisService.js

```diff
@@ -0,0 +1,115 @@
+const { ipcMain } = require('electron');
+const mpris = require('mpris-service');
+
+function executeMediaKey(win, key) {
+  win.webContents.executeJavaScript(`
+    window.electronConnector.emit('${key}')
+  `);
+}
+
+module.exports = (win, player) => {
+
+  const mprisPlayer = mpris({
+    name: 'headset',
+    identity: 'Headset',
+    canRaise: true,
+    supportedInterfaces: ['player'],
+    desktopEntry: 'headset',
+  });
+
+  mprisPlayer.playbackStatus = 'Stopped'
+  mprisPlayer.rate = 1+1e-15  //to avoid storing 1 as byte (nodejs dbus bug)
+  mprisPlayer.minimumRate = 1+1e-15
+  mprisPlayer.maximumRate = 1+1e-15
+  mprisPlayer.volume = 0.75
+
+  mprisPlayer.on('raise', () => {
+    win.show();
+  });
+
+  mprisPlayer.on('quit', () => {
+    exec('kill -9 $(pgrep headset) &> /dev/null')
+  });
+
+  mprisPlayer.on('rate', () => {
+    mprisPlayer.rate = 1+1e-15
+  });
+
+  mprisPlayer.on('playpause', () => {
+    if (mprisPlayer.playbackStatus == 'Playing' ||
+        mprisPlayer.playbackStatus == 'Paused') {
+      executeMediaKey(win, 'play-pause');
+    }
+  });
+
+  mprisPlayer.on('play', () => {
+    if (mprisPlayer.playbackStatus == 'Paused') {
+      executeMediaKey(win, 'play-pause');
+    }
+  });
+
+  mprisPlayer.on('pause', () => {
+    if (mprisPlayer.playbackStatus == 'Playing') {
+      executeMediaKey(win, 'play-pause');
+    }
+  });
+
+  mprisPlayer.on('next', () => {
+    if (mprisPlayer.playbackStatus == 'Playing' ||
+        mprisPlayer.playbackStatus == 'Paused') {
+          executeMediaKey(win, 'play-next');
+    }
+  });
+
+  mprisPlayer.on('previous', () => {
+    if (mprisPlayer.playbackStatus == 'Playing' ||
+        mprisPlayer.playbackStatus == 'Paused') {
+          executeMediaKey(win, 'play-previous')
+    }
+  });
+
+  mprisPlayer.on('volume', (volume) => {
+    if (volume > 1) { volume = 1 }
+    if (volume < 0) { volume = 0 }
+    player.webContents.send('win2Player', ['setVolume', volume*100]);
```

Comment:
There is a problem with this approach. `win2Player` are events that sent from the main window to the youtube player. In this case, It is mpris that is sending the events. The volume bar on the main window will not change. can you confirm?

Reply:
Yes, the volume bar doesn't change. I didn't know how to fix that. When sending `seekTo` events in the same way, the main windows does update the current track position, so maybe we could do something similar for the volume.
Another approach would be to execute some javascript directly on the main window that would let us move the volume slider, which will trigger a `setVolume` anyway.
For the first approach, I can't do anything about it. For the second one, just tell me the code to send, and I'll put it there.

Reply:
ok let's merge this and fix it later with a better integration